src/components/select/bl-select.ts

```diff
@@ -241,58 +330,74 @@ export default class BlSelect extends LitElement {
       'select-wrapper': true,
       'select-open': this.opened,
       'selected': this._selectedOptions.length > 0,
-      'invalid': this._isInvalid,
+      'invalid': !this.validity.valid,
+      'dirty': this.dirty
     })}
-      tabindex="-1"
+      @keydown=${this.handleKeydown}
     >
-      ${label} ${this.inputTemplate()} ${this.menuTemplate()} ${helpMessage} ${invalidMessage}
+      ${label}
+      ${this.inputTemplate()}
+      <div class="popover" tabindex="${ifDefined(this._isPopoverOpen ? undefined : '-1')}" @bl-select-option=${this._handleSelectOptionEvent}>
+        <slot></slot>
+      </div>
+      <div class="hint">${invalidMessage} ${helpMessage}</div>
     </div> `;
   }
 
-  private get _showPlaceHolder() {
-    if (this.label && !this.labelFixed) {
-      return !this._selectedOptions.length && this._isPopoverOpen;
+  private focusedOptionIndex = -1;
+
+  private handleKeydown(event: KeyboardEvent) {
+    if (this.focusedOptionIndex === -1 && event.code === 'Enter' || event.code === 'Space') {
+      this.togglePopover();
+    } else if (event.code === 'Escape') {
+      this.close();
+      event.preventDefault();
+    } else if (this._isPopoverOpen && ['ArrowDown', 'ArrowUp'].includes(event.code)) {
+      event.code === 'ArrowDown' && this.focusedOptionIndex++;
+      event.code === 'ArrowUp' && this.focusedOptionIndex--;
+
+      // Don't exceed array indexes
+      this.focusedOptionIndex = Math.max(
+        0,
+        Math.min(this.focusedOptionIndex, this.options.length - 1)
+      );
+
+      this.options[this.focusedOptionIndex].focus();
     }
-    return !this._selectedOptions.length;
+
   }
 
-  private _onClickSelectInput() {
+  private togglePopover() {
     this._isPopoverOpen ? this.close() : this.open();
   }
 
   private _handleSelectEvent() {
-    this._onBlSelect(this._selectedOptions);
+    this._onBlSelect(this._selectedOptions.map(option => ({
```

Comment:
Maybe not the right place, but it looks like I can change the options using keyboard even when the component is disabled. We have to check it at some place.

Reply:
Good catch, I'll fix it.

Reply:
Fixed it.